zeppelin-server/src/main/java/org/apache/zeppelin/service/NotebookService.java

```diff
@@ -344,19 +349,42 @@ public boolean runParagraph(String noteId,
     }
   }
 
-  public void runAllParagraphs(String noteId,
+  public boolean runAllParagraphs(String noteId,
+                      ServiceContext context,
+                      ServiceCallback<Paragraph> callback) throws IOException {
+    Note note = notebook.getNote(noteId);
+    if (note == null) {
+      callback.onFailure(new NoteNotFoundException(noteId), context);
+      return false;
+    }
+
+    List<Paragraph> paragraphs = note.getParagraphs();
+    List<Map<String, Object>> rawParagraphs = paragraphs.stream()
+            .map(p ->
+                    (Map<String, Object>) gson.fromJson(gson.toJson(p), new TypeToken<Map<String, Object>>(){}.getType()))
+            .collect(Collectors.toList());
+
+    return runAllParagraphs(
+            noteId,
+            rawParagraphs,
+            context,
+            callback
+    );
+  }
+
+  public boolean runAllParagraphs(String noteId,
```

Comment:
How about using the existing runAllParagraphs, and check whether `paragraphs`. If it is empty, then use the paragraph in backend, otherwise use the paragraphs from frontend.

Reply:
Addressed the comment. Please take a look updated code.